src/Games/NexusMods.Games.MountAndBlade2Bannerlord/LauncherManager/LauncherManagerFactory.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Concurrent;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Games.MountAndBlade2Bannerlord.Utils;
+
+namespace NexusMods.Games.MountAndBlade2Bannerlord.LauncherManager;
+
+public sealed class LauncherManagerFactory
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ConcurrentDictionary<string, LauncherManagerNexusMods> _instances = new();
+
+    public LauncherManagerFactory(IServiceProvider serviceProvider)
```

Comment:
Not sure if this will be needed in the long run, given you could invoke the constructor directly (and/or make the methods below static constructors).

I'm going to guess you have some state that takes a while to initialize, and are using this as a cache, in which case it's fine. Just make sure to note that in a comment if that's the case.

Reply:
Correct, the LauncherManager holds a state per game instance.
Yea, there was some PR I had a year or something ago that added a DI scope that was per game instance. In that case the LauncherManager could be a scoped service and this factory would not be needed

Reply:
I think it's fine either way; just note it down in a comment in there, and you'll be good.